File: app/src/main/java/com/liaudev/dicodingbpaa/data/DataSource.java
package com.liaudev.dicodingbpaa.data;

import androidx.lifecycle.LiveData;
import androidx.paging.Pager;

import com.liaudev.dicodingbpaa.data.local.entity.FavoriteEntity;
import com.liaudev.dicodingbpaa.data.local.entity.StoryEntity;
import com.liaudev.dicodingbpaa.data.remote.response.BaseResponse;
import com.liaudev.dicodingbpaa.data.remote.response.LoginResponse;
import com.liaudev.dicodingbpaa.data.remote.response.StoryResponse;
import com.liaudev.dicodingbpaa.vo.Resource;

import java.io.File;

/**
 * Created by Budiliauw87 on 2022-05-14.
 * budiliauw87.github.io
 * Budiliauw87@gmail.com
 */
public interface DataSource {
    LiveData<Resource<LoginResponse>> loginAccount(String username, String password);
    LiveData<Resource<BaseResponse>> registerAccount(String email, String name, String password);
    LiveData<FavoriteEntity> getFavoriteById(String id);
    LiveData<Resource<BaseResponse>> postStoryByGuest(String descreption, File filePhoto, double lat, double lon);
    LiveData<Resource<BaseResponse>> postStory(String descreption, File filePhoto, double lat, double lon);
    LiveData<Resource<StoryResponse>> getLocationStory(int page);
    Pager<Integer,StoryEntity> loadStory();
    void setFavorite(FavoriteEntity favorite);
    void deleteFavorite(String id);
}

File: app/src/main/java/com/liaudev/storydicoding/data/DataSource.kt
package com.liaudev.storydicoding.data

import androidx.lifecycle.LiveData
import com.example.mysubmission_intermediate.Model.UserModel
import com.liaudev.storydicoding.data.remote.response.LoginResponse
import com.liaudev.storydicoding.data.remote.response.RegisterResponse
import com.liaudev.storydicoding.vo.Resource
import kotlinx.coroutines.flow.Flow


/**
 * Created by Budiliauw87 on 2022-10-18.
 * budiliauw87.github.io
 * Budiliauw87@gmail.com
 */
interface DataSource {
    fun loginAccount(
        email: String,
        password: String
    ): Flow<Resource<LoginResponse>>

    fun registerAccount(
        email: String,
        username: String,
        password: String
    ): Flow<Resource<RegisterResponse>>

    fun getUser(): Flow<UserModel>
    suspend fun logOut()
    suspend fun saveUser(user: UserModel)
}